crates/containerd-shim-wasmtime/src/instance.rs

```diff
@@ -29,6 +28,7 @@ impl WasiConfig for DefaultConfig {
     fn new_config() -> Config {
         let mut config = wasmtime::Config::new();
         config.wasm_component_model(true); // enable component linking
+        config.async_support(true);
```

Comment:
What features are you looking to enable with the async support?

Reply:
I tried a sync implementation but got an error when running the demo app. If I remember correctly the error looked like [this](https://stackoverflow.com/questions/77696960/how-to-disable-async-in-wasmtimecomponent). I solved the problem by enabling async and using async linker and so on.